js/Modal.jsx

```diff
@@ -3,19 +3,25 @@ var ReactDOM = require("react-dom");
 
 var Modal = React.createClass({
 
-	getInitialState: function() {
+	getDefaultProps: function() {
 		return {
-		};
+			show: true
+		}
+	},
+
+	componentWillReceiveProps: function(props) {
+		this.modal(props);
 	},
 
 	componentDidMount: function() {
-		this.modal({show: true, keyboard: true});
+		this.modal(this.props);
 	},
 
 	componentWillUnmount: function() {
 		this.modal('hide');
 	},
 
+
 	modal: function(options) {
 		$(ReactDOM.findDOMNode(this)).modal(options);
```

Comment:
the ref syntax has changed a bit since when this component was written, and findDOMNode is no longer prefered way todo this. https://facebook.github.io/react/docs/top-level-api.html#reactdom.finddomnode

There is a reference in the DropDown.jsx that solves 

First create selector var https://github.com/silverstripe/deploynaut/blob/master/js/components/DropDown.jsx#L47

In the render of Modal, set a ref argument that looks something like this

https://github.com/silverstripe/deploynaut/blob/master/js/components/DropDown.jsx#L92

Now all places in this file that used to do `ReactDOM.findDOMNode(this)` can be replaced with `this.selector`, like this https://github.com/silverstripe/deploynaut/blob/master/js/components/DropDown.jsx#L62